### ISSUE TITLE
[amp story shopping] Key data in store service by page

### ISSUE THREAD

**Issue by processprocess** [ampproject/amphtml#37605]
### Description

Store service data needs to be keyed by page.
Page data should have a level of scope added, for example:
```
{
  'page1': // equal to the amp-story-page id attribute
    {
        'productId': 'art' // same structure as exists, just within the page key.
        ...
    },
    {
        'productId': 'vase' 
        ...
    },
  'page2': 
    {
        'productId': 'lamp' 
        ...
    },
    {
        'productId': 'flowers' 
        ...
    }
}
```

Currently data is not scoped by page.

Originally this was done so that product data could be shared across pages, but this does not work:
- If data for page 1 is configured on page 10, it will break. 
- Page level data is always required to render the experience.

For more context, see conversation [here](https://github.com/ampproject/amphtml/pull/37509#discussion_r794039673).

**Comment by jshamble** [comment]
Will solve with #37600

**Comment by processprocess** [comment]
> Will solve with #37600
Adding `Fixes #37600` to the description of #37600 will automatically link the issue to the PR. That way when the PR is merged the issue will close automatically! 
Please keep this in mind, it's a best practice for GitHub!
